providers/unboundid/src/test/org/sakaiproject/unboundid/UnboundidProviderTest.java

```diff
@@ -0,0 +1,665 @@
+/**********************************************************************************
+ * $URL$
+ * $Id$
+ ***********************************************************************************
+ *
+ * Copyright (c) 2003, 2004, 2005, 2006, 2007, 2008 The Sakai Foundation
+ *
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **********************************************************************************/
+
+package org.sakaiproject.unboundid;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Stack;
+
+import org.jmock.Mock;
+import org.jmock.cglib.MockObjectTestCase;
+import org.jmock.core.Invocation;
+import org.jmock.core.Stub;
+import org.sakaiproject.unboundid.EidDerivedEmailAddressHandler;
+import org.sakaiproject.unboundid.EidValidator;
+import org.sakaiproject.unboundid.InvalidEmailAddressException;
+import org.sakaiproject.unboundid.LdapAttributeMapper;
+import org.sakaiproject.unboundid.LdapEntryMapper;
+import org.sakaiproject.unboundid.LdapUserData;
+import org.sakaiproject.unboundid.UnboundidDirectoryProvider;
+import org.sakaiproject.user.api.UserEdit;
+
+import com.unboundid.ldap.sdk.migrate.ldapjdk.LDAPConnection;
+import com.unboundid.ldap.sdk.migrate.ldapjdk.LDAPEntry;
+import com.unboundid.ldap.sdk.migrate.ldapjdk.LDAPException;
+import com.unboundid.ldap.sdk.migrate.ldapjdk.LDAPSearchResults;
+
+/**
+ * 
+ * @author dmccallum@unicon.net
+ *
+ */
+public class UnboundidDirectoryProviderTest extends MockObjectTestCase {
```

Comment:
This test is failing as it doesn't match the filename
The test is failing here with an incorrectly named file 
[ERROR] COMPILATION ERROR : 
[ERROR] /home/travis/build/sakaiproject/sakai/providers/unboundid/src/test/org/sakaiproject/unboundid/UnboundidProviderTest.java:[51,8] class UnboundidDirectoryProviderTest is public, should be declared in a file named UnboundidDirectoryProviderTest.java